src/App.tsx

```diff
@@ -1,54 +1,123 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [visible, setVisible] = useState(false);
+
+  // eslint-disable-next-line
+  const applyQuery = useCallback(
+    debounce((value: string) => setAppliedQuery(value), 1000),
+    [],
+  );
+
+  const handleOnClick = (eventOnClick: React.MouseEvent, person: Person) => {
+    eventOnClick.preventDefault();
+
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setAppliedQuery(person.name);
+    setVisible(false);
+  };
+
+  const onBlurDelay = () => {
+    setTimeout(() => setVisible(false), 100);
+  };
+
+  const handleReset = () => {
+    setSelectedPerson(null);
+    setAppliedQuery('');
+    setQuery('');
+  };
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const preparedPeople = (people: Person[], queryFilter: string) => {
```

Comment:
`preparedPeople` better to name it `preparePeople`, without ed at the end, because it's function.

Reply:
Correct. Fix, please.